pkg/distributor/distributor.go

```diff
@@ -301,7 +327,7 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	// check each sample and discard if outside limits.
 	validatedTimeseries := make([]client.PreallocTimeseries, 0, len(req.Timeseries))
 	keys := make([]uint32, 0, len(req.Timeseries))
-	numSamples := 0
+	numSamples = 0
```

Comment:
Does `numSamples` need to be recounted? Is there any way that it would differ from the above calculation?